server/remote_cache/cachetools/cachetools.go

```diff
@@ -932,15 +943,21 @@ func streamTree(ctx context.Context, casClient repb.ContentAddressableStorageCli
 			SendCachedSubtreeDigests: sendCachedSubtreeDigests,
 		})
 		if err != nil {
-			return nil, nil, err
+			if status.IsNotFoundError(err) {
```

Comment:
nit: might be slightly cleaner if you did this check inside the retry.Do in streamTreeWithRetries rather than repeating it inside streamTree